src/main/kotlin/org/ionproject/integration/builder/implementations/IselTimetableTeachersBuilder.kt

```diff
@@ -200,15 +207,22 @@ class IselTimetableTeachersBuilder() : ITimetableTeachersBuilder<RawData> {
 
     private fun populateCourseDetails(text: String): Triple<String, String, String> {
         val firstIndex = text.indexOf('(')
-        val secondIndex = text.indexOf(')') + 1
 
         return Triple(
-            text.substring(0, firstIndex),
-            text.substring(firstIndex, secondIndex),
-            text.substring(secondIndex)
+            text.substring(0, text.indexOf('[')),
+            text.substring(text.lastIndexOf(')') + 1),
+            text.substring(firstIndex + 1, text.indexOf(')', firstIndex))
         )
     }
 
+    private fun getDescription(classType: String) = when (classType) {
+        "T" -> "Aulas Teóricas de "
+        "P" -> "Aulas Práticas de "
+        "L" -> "Aulas Laboratório de "
+        "T/P" -> "Aulas Teórico-práticas de "
+        else -> ""
+    }
+
```

Comment:
@pmhsfelix Do you agree with this values?

Reply:
Where is this used? I think, so for the moment being. We probably need to revisit this but that will be done "next year".